test/integration/stats/applications_test.rb

```diff
@@ -8,37 +8,33 @@ def setup
     @service = @provider.default_service
     @plan = FactoryBot.create(:simple_application_plan, issuer: @service)
     @application = FactoryBot.create(:simple_cinstance, plan: @plan)
+    @member = FactoryBot.create(:member, account: @provider, allowed_sections: %w[partners plans], state: 'active')
 
     host! @provider.external_admin_domain
-    login_provider @provider
+    login_provider @provider, user: @member
   end
 
   test '#show nonexistent application does not check permissions' do
-    User.any_instance.expects(:has_access_to_all_services?).never
```

Comment:
Throughout the tests, I removed stubbing `has_access_to_all_services?` and opted for actually setting the proper permissions for the member. I think it is less confusing, and more reliable.

I might eventually remove this method...

`all_services_allowed?` could be a good replacement for it, as it essentially returns the same - but in a correct way :) (i.e. not returning `true` for members that don't have **ANY** permissions). However, stubbing it is not really useful, because the `Service.permitted_for` scope doesn't rely on it.

That's why I actually think that rather than stub the methods, we can just set the proper state of the object.

Reply:
+1, but I see you have updated `has_access_to_all_services?` to work in the correct way already so maybe this comment is a little dated. Or not?

Reply:
Yes, it is :D 

I'll update the comment, I refactored the code, and now `has_access_to_all_services?` actually does the same that this new (and now removed) `all_services_allowed?` method did.